src/sidebar/config/host-config.js

```diff
@@ -78,35 +78,47 @@ export function hostPageConfig(window) {
   // It is assumed we should expand this list and coerce and eventually
   // even validate all such config values.
   // See https://github.com/hypothesis/client/issues/1968
+
+  /** @type {Record<string, (value: unknown) => unknown>} */
   const coercions = {
     openSidebar: toBoolean,
+
+    /** @param {unknown} value */
     requestConfigFromFrame: value => {
       if (typeof value === 'string') {
         // Legacy `requestConfigFromFrame` value which holds only the origin.
         return value;
       }
-      const objectVal = toObject(value);
+      const objectVal =
+        /** @type {{ origin: unknown, ancestorLevel: unknown }} */ (
+          toObject(value)
+        );
       return {
         origin: toString(objectVal.origin),
         ancestorLevel: toInteger(objectVal.ancestorLevel),
       };
     },
   };
 
-  return Object.keys(config).reduce((result, key) => {
```

Comment:
I rewrote this `Object.keys(config).reduce(..., {})` loop as a way to build up an object containing sanitized config values into a for loop. This is IMO easier to read because the type of the result we are building up is declared at the top rather than the bottom of the loop.